src/tunnelbridge_cmd.cpp

```diff
@@ -995,6 +1036,23 @@ static CommandCost DoClearBridge(TileIndex tile, DoCommandFlag flags)
 		}
 		DirtyCompanyInfrastructureWindows(owner);
 
+		DeleteNewGRFInspectWindow(GSF_BRIDGES, tile);
+		DeleteNewGRFInspectWindow(GSF_BRIDGES, endtile);
+
+		Bridge *b = Bridge::Get(GetBridgeIndex(tile));
+		Axis axis = b->GetAxis();
+
+		auto *list = &_bridge_index[axis][axis == AXIS_X ? TileY(tile) : TileX(tile)];
```

Comment:
I see this logic in a few places. Consider making a function in `bridge_map.cpp` that returns an appropriate iterator instead, or even writing a wrapper iterator class.

Reply:
c903c31 could fix it.